### PR TITLE
Feature/add rpc generic throw exception

### PR DESCRIPTION
**Pull request by rickey17**
### Motivation:

解决一个issue问题

https://github.com/sofastack/sofa-rpc/issues/1258

### Modification:

增加sofaconfig配置sofa.rpc.generic.throw.exception，可以解决这个问题；不增加的话，保持之前的行为不变化

### CODE REVIEW

**Comment by OrezzerO** [comment]
@rickey17 Some test case failed, please take a look.

File: codec/codec-sofa-hessian/src/main/java/com/alipay/sofa/rpc/codec/sofahessian/serialize/CustomThrowableGenericDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ */
+public class CustomThrowableGenericDeserializer {
+
+    private static boolean        GENERIC_THROW_EXCEPTION = "true".equalsIgnoreCase(System
+                                                              .getProperty("rpc.generic.throw.exception"));
+
+    private static final String[] THROWABLE_FIELDS        = new String[] { "cause", "detailMessage", "stackTrace",
+                                                          "suppressedExceptions" };
+
+    public static Object judgeCustomThrowable(Object appObject) {
```
Comment (EvenLjj) [inline codec/codec-sofa-hessian/src/main/java/com/alipay/sofa/rpc/codec/sofahessian/serialize/CustomThrowableGenericDeserializer.java:34]: This method only identifies custom exceptions under generalized calls，so can modify appropriate method name.

File: codec/codec-sofa-hessian/src/main/java/com/alipay/sofa/rpc/codec/sofahessian/serialize/CustomThrowableGenericDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+package com.alipay.sofa.rpc.codec.sofahessian.serialize;
+
+import com.alipay.hessian.generic.model.GenericObject;
+
+/**
+ *
+ * @author xingqi
+ * @version : CustomThrowableGenericDeserialize.java, v 0.1 2022年10月18日 5:02 PM xingqi Exp $
+ */
+public class CustomThrowableGenericDeserializer {
```
Comment (EvenLjj) [inline codec/codec-sofa-hessian/src/main/java/com/alipay/sofa/rpc/codec/sofahessian/serialize/CustomThrowableGenericDeserializer.java:26]: This class is less like a deserialized class, but more like a util class.

File: codec/codec-sofa-hessian/src/main/java/com/alipay/sofa/rpc/codec/sofahessian/serialize/CustomThrowableGenericDeserializer.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    private static boolean        GENERIC_THROW_EXCEPTION = "true".equalsIgnoreCase(System
```
[the remark this answers is not in the document]
Comment (rickey17) [reply codec/codec-sofa-hessian/src/main/java/com/alipay/sofa/rpc/codec/sofahessian/serialize/CustomThrowableGenericDeserializer.java:28]: ok

File: codec/codec-sofa-hessian/src/main/java/com/alipay/sofa/rpc/codec/sofahessian/serialize/CustomThrowableGenericDeserializer.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            if (!((GenericObject) appObject).hasField(field)) {
+                return appObject;
+            }
+        }
+        return new RuntimeException(
+            "occur business exception, but type=" + ((GenericObject) appObject).getType() +
+                " class is not found, error: " + appObject);
+    }
+
+    public static void setGenericThrowException(boolean enabled) {
```
[the remark this answers is not in the document]
Comment (rickey17) [reply codec/codec-sofa-hessian/src/main/java/com/alipay/sofa/rpc/codec/sofahessian/serialize/CustomThrowableGenericDeserializer.java:51]: ok

File: codec/codec-sofa-hessian/src/main/java/com/alipay/sofa/rpc/codec/sofahessian/serialize/CustomThrowableGenericDeserializer.java (hunk 1 above)
Comment (rickey17) [reply codec/codec-sofa-hessian/src/main/java/com/alipay/sofa/rpc/codec/sofahessian/serialize/CustomThrowableGenericDeserializer.java:34]: ok

File: codec/codec-sofa-hessian/src/main/java/com/alipay/sofa/rpc/codec/sofahessian/serialize/CustomThrowableGenericDeserializer.java (hunk 2 above)
Comment (rickey17) [reply codec/codec-sofa-hessian/src/main/java/com/alipay/sofa/rpc/codec/sofahessian/serialize/CustomThrowableGenericDeserializer.java:26]: yes

**Review by EvenLjj — APPROVED** [review]
LGTM

**Review by OrezzerO — APPROVED** [review]
LGTM

**Pull request merged**